### PR TITLE
fix: Get user first name and last name from Apple

### PR DESCRIPTION
**Pull request by JiggyDown**
Currently we can't get user first name and last name from Apple as the names directly to the callback URL, as opposed to embedding them in the ID token like other OIDC providers. This PR makes sure we handle that behavior properly.

## Related issue(s)

N/A

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [x] I have added or changed [the documentation](docs/docs).

## Further Comments

This PR requires an E2E test, but I can't find an existing one for OIDC. Can you give me some pointers on how the test should be done?

### CODE REVIEW

File: selfservice/strategy/oidc/provider_apple.go [hunk 1]
```diff
@@ -101,3 +104,37 @@ func (a *ProviderApple) AuthCodeURLOptions(r ider) []oauth2.AuthCodeOption {
...
+
+func (a *ProviderApple) Claims(ctx context.Context, exchange *oauth2.Token, query url.Values) (*Claims, error) {
+	claims, err := a.ProviderGenericOIDC.Claims(ctx, exchange, query)
+	if err != nil {
+		return claims, err
+	}
+
+	// https://developer.apple.com/documentation/sign_in_with_apple/sign_in_with_apple_js/configuring_your_webpage_for_sign_in_with_apple#3331292
+	// First name and last name are passed only once in an additional parameter to the redirect URL.
+	// There's no way to make sure they haven't been tampered with. Blame Apple.
```
Comment (aeneasr) [inline selfservice/strategy/oidc/provider_apple.go:116]: lol what

File: selfservice/strategy/oidc/provider_apple.go [hunk 2]
```diff
@@ -101,3 +104,37 @@ func (a *ProviderApple) AuthCodeURLOptions(r ider) []oauth2.AuthCodeOption {
...
+	var user struct {
+		Name *struct {
+			FirstName *string `json:"firstName"`
+			LastName  *string `json:"lastName"`
+		} `json:"name"`
+	}
+	if err := json.Unmarshal([]byte(query.Get("user")), &user); err == nil {
```
Comment (aeneasr) [inline selfservice/strategy/oidc/provider_apple.go:123]: Could you please add a test for this?

**Review by aeneasr — CHANGES_REQUESTED** [review]
Thank you, this looks great! Since this is very difficult to test, could you please create a short screencast of this feature working and post it here?

Also can you confirm that https://www.ory.sh/docs/kratos/guides/sign-in-with-github-google-facebook-linkedin#apple is correct? There were a few people who had issues setting up Apple / cc @vinckr

File: selfservice/strategy/oidc/provider_apple.go (hunk 2 above)
Comment (aeneasr) [reply selfservice/strategy/oidc/provider_apple.go:123]: @JiggyDown if you could add a small test for this, we can merge this right away :)

Comment (JiggyDown) [reply selfservice/strategy/oidc/provider_apple.go:129]: Test added. Thank you for your patience!

**Review by aeneasr — APPROVED** [review]
Awesome, thank you for making Ory better!

**Comment by vinckr** [comment]
Hello @JiggyDown 
Congrats on merging your first PR in Ory 🎉 !
Your contribution will soon be helping secure millions of identities around the globe 🌏. 
As a small token of appreciation we send all our first time contributors a gift package to welcome them to the community. 
[Please drop me an email and I will forward you the form to claim your Ory swag!](mailto:vincent@ory.sh)

**Pull request merged**